tests/cloud-init

```diff
@@ -56,5 +56,31 @@ rm ./profile-key ./profile-key.pub ./additional-key ./additional-key.pub
 lxc profile unset default cloud-init.user-data
 lxc delete -f c1
 
+echo "==> Use the NoCloud datasource by using a disk device with a \"cloud-init:config\" source"
+lxc init ubuntu:n vm1 --vm -c cloud-init.user-data="$(cat <<EOF
+#cloud-config
+users:
+  - name: "{{ ds.meta_data.user_user }}"
+EOF
+)"
+
+if hasNeededAPIExtension "cloud_init_ssh_keys"; then
+  lxc config set vm1 user.user=user
+else
+  lxc config set vm1 user.metadata="$(cat <<EOF
+user_user: user
+EOF
+)"
+fi
+lxc config device add vm1 config disk source=cloud-init:config
+lxc start vm1
+waitInstanceBooted c1
+lxc exec vm1 -- cloud-init status --wait --long | grep "detail: DataSourceNoCloud"
```

Comment:
I'd probably do it in 2 steps. First `--wait --long` alone, then grep'ing for the expected output. It would make it easier to debug IMHO.